plugins/woocommerce-blocks/assets/js/base/components/cart-checkout/form/address-2-field.tsx

```diff
@@ -0,0 +1,98 @@
+/**
+ * External dependencies
+ */
+import { ValidatedTextInput } from '@woocommerce/blocks-components';
+import { AddressFormValues, ContactFormValues } from '@woocommerce/settings';
+import {
+	useCallback,
+	useEffect,
+	useMemo,
+	useState,
+	Fragment,
+} from '@wordpress/element';
+import { objectHasProp } from '@woocommerce/types';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { AddressFieldProps } from './types';
+
+const Address2Field = < T extends AddressFormValues | ContactFormValues >( {
+	field,
+	fieldProps,
+	fieldsRef,
+	onChange,
+	values,
+}: AddressFieldProps< T > ): JSX.Element => {
+	const hasFieldValue = useMemo( () => {
+		return objectHasProp( values, 'address_2' ) && values.address_2 !== '';
+	}, [ values ] );
+
+	const [ hasFieldBeenModified, setHasFieldBeenModified ] = useState( false );
+
+	const isFieldRequired = field ? field.required : false;
+
+	const [ isFieldVisible, setFieldVisible ] = useState(
+		hasFieldValue || isFieldRequired
+	);
```

Comment:
This would change to only run once, which is what we need here, via 
```suggestion
	const [ isFieldVisible, setFieldVisible ] = useState(
		() => hasFieldValue || isFieldRequired
	);
```

Reply:
That's a great suggestion, @senadir. I wasn't aware of the concept of lazy initialization yet. Thanks for sharing this. 🙌